src/components/Visualization/Visualization.js

```diff
@@ -144,15 +139,10 @@ export class UnconnectedVisualization extends Component {
 
         this.props.addMetadata(forMetadata)
 
-        if (
-            !responses.some((response) => response.rows && response.rows.length)
-        ) {
-            if (this.props.visualization.type === VIS_TYPE_OUTLIER_TABLE) {
-                throw new NoOutliersError()
-            }
-
-            throw new EmptyResponseError()
-        }
+        ensureAnalyticsResponsesContainData(
```

Comment:
I see that this wasn't previously in a try/catch either. Does this ever throw? What does the user experience in this case?

Reply:
I worked on this quite a while ago, but I am pretty sure it is like this (quite complicated):
1. `onResponsesReceived` is defined in `Visualization` and calls `ensureAnalyticsResponsesContainData`. It is passed as a prop to `PluginWrapper`.
2. It is and then called by the `PluginWrapper` when a response comes in: it is called by `doFetchData` which in turn is called in `doFetchAll`.
3. `doFetchAll()` returns a Promise and the catch method of this promise is called (`doFetchAll().catch(..)`). In the catch calback `onError()` is being called with the error. So if `ensureAnalyticsResponsesContainData()` throws an error this will be the parameter passed to `onError(error)`
4. `onError` itself is another prop called by the `PluginWrapper` but defined in the `Visualization`. It ends up calling `this.props.setLoadError(error)` and this will ensure the `StartScreen` is rendered with the appropriate message.